kart/tabular/export.py

```diff
@@ -0,0 +1,306 @@
+
+import click
+from osgeo import gdal, ogr
+
+from kart.cli_util import (
+    call_and_exit_flag,
+    KartCommand,
+)
+from kart.completion_shared import repo_path_completer
+from kart.crs_util import make_crs
+from kart.exceptions import CrsError
+
+
+def is_writable_vector_format(driver):
+    m = driver.GetMetadata()
+    return m.get("DCAP_VECTOR") == "YES" and m.get("DCAP_CREATE") == "YES"
+
+
+def list_export_formats(ctx):
+    """List the supported export formats as reported by gdal."""
+
+    for i in range(ogr.GetDriverCount()):
+        d = ogr.GetDriver(i)
+        if is_writable_vector_format(d):
+            click.echo(f"    {d.GetName()}: {d.GetMetadataItem(gdal.DMD_LONGNAME)}")
+
+
+def get_driver_by_shortname(shortname):
+    shortname = shortname.upper()
+    for i in range(ogr.GetDriverCount()):
+        d = ogr.GetDriver(i)
+        if d.GetName().upper() == shortname:
+            if not is_writable_vector_format(d):
+                raise click.UsageError(
+                    f"Driver {shortname} does not support writing vectors"
+                )
+            return d
+    raise click.UsageError(f"Destination format {shortname} not recognized")
+
+
+def get_driver_by_ext(filename):
+    if "." not in filename:
+        raise click.UsageError(
+            f"Destination format for filename {filename} not recognized"
+        )
+
+    ext = os.path.splitext(filename)[1][1:].upper()
+
+    for i in range(ogr.GetDriverCount()):
+        d = ogr.GetDriver(i)
+        exts = d.GetMetadataItem(gdal.DMD_EXTENSIONS)
+        if is_writable_vector_format(d) and exts:
+            exts = exts.upper().split(" ")
+            if ext in exts:
+                return d
+    raise click.UsageError(f"Destination format for filename {filename} not recognized")
+
+
+def get_driver(destination_spec):
+    """
+    Given a destination like target.gpkg, GPKG:target.gpkg, postgresql://x/y/z,
+    returns the driver to use and the target that the driver should write to.
+    """
+    match = re.match(r'([^:]+)://', destination_spec)
+    if match:
+        return get_driver_by_shortname(match.group(1)), destination_spec
+    if ":" in destination_spec:
+        shortname, destination = destination_spec.split(":", maxsplit=1)
+        return get_driver_by_shortname(shortname), destination
+    return get_driver_by_ext(destination_spec), destination_spec
+
+
+@click.command(
+    "table-export",
+    cls=KartCommand,
+    hidden=True,
+    context_settings=dict(ignore_unknown_options=True),
+)
+@click.pass_context
+@call_and_exit_flag(
+    "--list-formats",
+    callback=list_export_formats,
+    help="List available export formats, and then exit",
+)
+@click.option(
+    "--ref", default="HEAD", help="The revision at which to export the dataset."
+)
+@click.option(
+    "--layer",
+    help="Name of the layer to create inside the given destination. Defaults to the dataset path.",
+)
+@click.option(
+    "--layer-creation-option",
+    "-lco",
```

Comment:
do these short options work with click? Cool if so - but I kind of expected it to condence multiple short single-leter options into one argument (`ls -lha` style)

Reply:
They do work! Perhaps the other sort `-lha` don't work? Or perhaps it's really clever